lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -528,14 +530,31 @@ func processAnonymousVolumes(h *string, volumes map[string]struct{}, client *cli
 	return volumeFields, nil
 }
 
+// processSpecifiedVolumes parses fields for volumes specified explicitly during a
+// create/run (such as -v volume-name:/dir) and creates them if they don't already exist.
 func processSpecifiedVolumes(volumes []string) ([]volumeFields, error) {
 	var volumeFields []volumeFields
+	vol := &Volume{}
+
 	for _, v := range volumes {
 		fields, err := processVolumeParam(v)
 		log.Infof("Processed specified volume arguments: %#v", fields)
 		if err != nil {
 			return volumeFields, err
 		}
+
+		// Error out if the intended volume is a directory on the client filesystem.
+		if strings.HasPrefix(fields.ID, "/") {
+			return nil, fmt.Errorf("%s does not support mounting directories as a data volume.", ProductName())
+		}
+
+		driverArgs := make(map[string]string)
+		driverArgs["flags"] = fields.Flags
+		_, err = vol.VolumeCreate(fields.ID, "vsphere", driverArgs, nil)
+		if err == nil {
```

Comment:
That is correct. This implicitly handles the case when a volume with the same name already exists and we don't need to create it - `VolumeCreate` will not create that volume and return an error. I can add error handling for the `VolumeCreate` call to differentiate b/w the error scenarios, but it either succeeds or returns an error with `http.StatusInternalServerError` in all but one cases.